packages/edit-site/src/components/header/document-actions/index.js

```diff
@@ -73,10 +73,15 @@ export default function DocumentActions( {
 } ) {
 	const { label } = useSecondaryText();
 
-	// The title ref is passed to the popover as the anchorRef so that the
-	// dropdown is centered over the whole title area rather than just one
-	// part of it.
-	const titleRef = useRef();
+	// Use internal state instead of a ref to make sure that the component
+	// re-renders when then anchor's ref updates.
+	const [ popoverAnchor, setPopoverAnchor ] = useState();
+	const titleWrapperCallbackRef = useCallback( ( node ) => {
+		// Use the title wrapper as the popover anchor so that the dropdown is
+		// centered over the whole title area rather than just one part of it.
+		// Fall back to `undefined` in case the ref is `null`.
+		setPopoverAnchor( node ?? undefined );
+	}, [] );
```

Comment:
@ciampo, while I understand what this does. However, I'm not sure I fully understand why it's needed here 😅  Are there cases when the regular ref isn't set during the render?

Reply:
I haven't tested this component extensively to see if there's any practical cases, but in theory:

- `titleRef` would be `undefined` during the component's initial render
- `titleRef` would update to the correct ref, but that wouldn't cause the component to re-render
- if no other external factors caused the component to re-render, the `Popover` component would not receive an updated `anchor`

For this specific component, it would be entirely possible that the component re-renders multiple times for different reasons (i.e. `isLoading` changes from `true` to `false`), but I still believe that this code change is a good idea:

- it makes the component more resiliant to future changes
- it shows the correct pattern around making sure the `anchor` value is never stale, in case any other developer took it as an example

Reply:
Thanks, @ciampo.

I think that's a good argument, and we can justify an extra render.